alerting/src/test/kotlin/org/opensearch/alerting/util/clustermetricscoordinatortest/ClusterMetricsCoordinatorIT.kt

```diff
@@ -0,0 +1,209 @@
+import org.opensearch.client.ResponseException
+import org.opensearch.common.xcontent.XContentFactory.jsonBuilder
+import org.opensearch.common.xcontent.XContentType
+import org.opensearch.rest.RestStatus
+import java.time.Instant
+import java.time.temporal.ChronoUnit
+import java.util.*
+import kotlin.collections.ArrayList
+class ClusterMetricsCoordinatorIT : AlertingRestTestCase() {
+    @Before
+    fun setup() {
+        // When setting up the tests, change the execution frequency and history max age settings to 1 minute and 10 minutes from
+        // 15 minutes and 7 days.
+        generateData()
+        val response = getSettings()
+        val persistentMap = response["persistent"] as Map<String, Any>
+        val executionFrequency = persistentMap["plugins.alerting.cluster_metrics.execution_frequency"].toString()
+        val storageTime = persistentMap["plugins.alerting.cluster_metrics.metrics_history_max_age"].toString()
+        assertEquals(executionFrequency, "1m")
+        assertEquals(storageTime, "10m")
+    }
+
+    fun `test checkName`() {
+        // WHEN + THEN, check whether the created index exists and has the name '.opendistro-alerting-cluster-metrics'
+        val index = ClusterMetricsVisualizationIndex.CLUSTER_METRIC_VISUALIZATION_INDEX
+        val response = client().makeRequest("HEAD", index)
+        assertEquals("Index $index does not exist.", RestStatus.OK, response.restStatus())
+    }
+
+    fun `test numberDocs`() {
+        // Check that the total number of documents found is divisible by the total number of metric types.
+        val response = getResponse()
+        val hits = getHits(response)
+        val numberOfDocsFound = (hits["total"]?.get("value") as Int)
+        val size = ClusterMetricsDataPoint.MetricType.values().size
+        assertEquals((numberOfDocsFound.mod(size)), 0)
+        val docs = hits["hits"] as ArrayList<Map<String, Any>>
+
+        // check that each of the metric types has a unique timestamp, and number of timestamps must be equal to total docs divided by 7
+        // expect that there should only be one document created for each metric type at a time.
+        val mapCheck = hashMapOf<String, Set<String>>()
+        ClusterMetricsDataPoint.MetricType.values().forEach { mapCheck[it.metricName] = mutableSetOf() }
+        for (doc in docs) {
+            val source = doc["_source"] as Map<String, Map<String, Any>>
+            val metricType = source.keys.first()
+            try {
+                ClusterMetricsDataPoint.MetricType.valueOf(metricType.uppercase(Locale.getDefault()))
+                mapCheck[metricType] = mapCheck[metricType]?.plus(source[metricType]?.get("timestamp")) as Set<String>
+            } catch (e: java.lang.IllegalArgumentException) {
+                logger.info("Key does not exist in the enum class.")
+            }
+        }
+        assertEquals(mapCheck.values.toSet().size, 1)
+    }
+
+    fun `test deleteDocs`() {
```

Comment:
You'll want to make the test name a bit more descriptive so that the test case is more obvious.